src/rpc/blockchain.cpp

```diff
@@ -2234,6 +2235,39 @@ class BlockFiltersScanReserver
     }
 };
 
+static bool CheckBlockFilterMatches(const CBlockIndex* pindex, const GCSFilter::ElementSet& needles, const CChainParams& chainparams)
+{
+    CBlock block;
+    if (!ReadBlockFromDisk(block, pindex, chainparams.GetConsensus())) {
+        return false;
```

Comment:
Here and below when reading the undo data, if the read fails then this returns false and the block is recorded as a false positive and won't be returned by `scanblocks`. These should instead throw a `JSONRPCError` because the results will not be correct.